src/System.Security.Cryptography.X509Certificates/tests/CertificateCreation/CertificateRequestChainTests.cs

```diff
@@ -201,6 +218,85 @@ public static void ChainCertRequirements(bool useIntermed, bool? isCA, X509KeyUs
             }
         }
 
+        [Fact]
+        public static void BuildChainUntrustedRoot()
+        {
+            using (RSA rootPrivKey = RSA.Create(3072))
+            using (RSA intermed1PrivKey = RSA.Create(2048))
+            using (RSA leafKey = RSA.Create(1536))
+            using (RSA leafPubKey = RSA.Create(leafKey.ExportParameters(false)))
+            {
+                leafPubKey.ImportParameters(leafKey.ExportParameters(false));
+                CreateCertificates(rootPrivKey, intermed1PrivKey, intermed1PrivKey, leafPubKey, BuildAndTestChainLocal, false);
+            }
+
+            void BuildAndTestChainLocal(
+                X509Certificate2 rootCert,
+                X509Certificate2 intermed1Cert,
+                X509Certificate2 intermed2Cert,
+                X509Certificate2 leafCert,
+                bool rootExtraStore)
+            {
+                using (X509Chain chain = new X509Chain())
+                {
+                    chain.ChainPolicy.ExtraStore.Add(rootCert);
+                    chain.ChainPolicy.RevocationMode = X509RevocationMode.NoCheck;
+                    chain.ChainPolicy.VerificationTime = DateTimeOffset.UtcNow.ToLocalTime().DateTime;
+                    chain.ChainPolicy.TrustMode = X509ChainTrustMode.CustomRootTrust;
+                    chain.ChainPolicy.CustomTrustStore.Add(intermed2Cert);
+
+                    RunChain(chain, leafCert, false, "Initial chain build");
+
+                    try
+                    {
+                        Assert.Equal(LeafDN, chain.ChainElements[0].Certificate.Subject);
+                        Assert.Equal(Intermed2DN, chain.ChainElements[1].Certificate.Subject);
+                        Assert.Equal(3, chain.ChainElements.Count);
+                        Assert.Equal(X509ChainStatusFlags.UntrustedRoot, chain.AllStatusFlags());
+                    }
+                    finally
+                    {
+                        DisposeChainCerts(chain);
+                    }
+                }
+            }
+        }
+
+        [Fact]
+        public static void CustomStoreWithPublicCertificate()
```

Comment:
The name on this one seems a little off, too.  `public static void SystemTrustedCertificateWithCustomRootTrust(bool trustCorrectRoot)` seems better.  If this one doesn't need any dynamic elements it can go straight into ChainTests.cs